cdc/owner/schema.go

```diff
@@ -102,12 +120,17 @@ func (s *schemaWrap4Owner) AllTableNames() []model.TableName {
 }
 
 func (s *schemaWrap4Owner) HandleDDL(job *timodel.Job) error {
-	if job.BinlogInfo.FinishedTS <= s.ddlHandledTs {
+	// we use schemaVersion to check if there are any already executed DDL received again,
+	// the newer DDL job has bigger schemaVersion.
+	if job.BinlogInfo.FinishedTS <= s.ddlHandledTs || job.BinlogInfo.SchemaVersion <= s.schemaVersion {
 		log.Warn("job finishTs is less than schema handleTs, discard invalid job",
 			zap.String("namespace", s.id.Namespace),
 			zap.String("changefeed", s.id.ID),
 			zap.Stringer("job", job),
-			zap.Any("ddlHandledTs", s.ddlHandledTs))
+			zap.Any("ddlHandledTs", s.ddlHandledTs),
+			zap.Int64("schemaVersion", s.schemaVersion),
+			zap.Int64("jobSchemaVersion", job.BinlogInfo.SchemaVersion),
+		)
 		return nil
```

Comment:
Code duplication?

Reply:
no, there is two place about this code in codebase